qutip/core/data/ptrace.pyx

```diff
@@ -82,8 +87,12 @@ cdef inline void _i2_k_t(idxint N, idxint[:, ::1] tensor_table, idxint out[2]):
 
 
 cpdef CSR ptrace_csr(CSR matrix, object dims, object sel):
-    _check_shape(matrix)
-    dims, sel = _prepare_inputs(dims, sel)
+    dims = np.atleast_1d(dims).astype(idxint_dtype).ravel()
+    sel = np.atleast_1d(sel).astype(idxint_dtype)
+    sel.sort()
```

Comment:
Could we move these into `_check_inputs` somehow? My concern is that it is an error to call `_check_inputs` without running these lines (e.g. `dims` might be very nested) and it would be easy for us to forget this later.  It would require returning `dims` and `sel` and maybe changing the function name to `_parse_inputs` or something that suggests that inputs are parsed and modified.